src/puptoo/process/__init__.py

```diff
@@ -92,10 +92,12 @@ def extract(msg, extra, remove=True):
     'insights_id', 'subscription_manager_id', 'satellite_id',
     'bios_uuid', 'ip_addresses', 'mac_addresses', 'fqdn'
 ]
+CANONICAL_ID_FACTS = ['provider_id', 'subscription_manager_id', 'insights_id']
 
 
 def validateCanonicalFacts(facts):
-    if (all(key in facts for key in PROVIDER)
-            or not any(key in facts for key in PROVIDER)) and any(key in facts for key in FACTS_EXCEPT_PROVIDER):
+    if ((all(key in facts for key in PROVIDER) or not any(key in facts for key in PROVIDER))
+            and any(key in facts for key in CANONICAL_ID_FACTS)
+            and any(key in facts for key in FACTS_EXCEPT_PROVIDER)):
```

Comment:
Per this logic, the `FACTS_EXCEPT_PROVIDER` is not required at all. It can be removed.